### DISCUSSION TITLE
Expose `Entry` in MatrixInput.js

### DISCUSSION THREAD

**Discussion by vandres** [craftcms/cms#14878] [Ideas] [answered]
I am trying to extend Matrix a little so it is closer to Neo. Currently, I am trying to allow Duplication/Clone of inline editable Matrix entries.

To make that work, I think, I need access to the `Entry`.

Would it be possible to expose that by e.g. `Craft.MatrixEntry` or just by giving a factory method in `Craft.MatrixInput`, like `Craft.MatrixInput.createEntry($container)`. I think the ladder would need less changes.

**Comment by brandonkelly** [comment, marked as answer]
That class has been renamed to `Craft.MatrixInput.Entry` for Craft 5.1 (d9bf77ec4618a7ada1f362e319192594a47af838). Sounds like that’ll work for you :)

**Reply by vandres** [reply]
That'll do, thank you very much. Is any of the theses features on your roadmap for Matrix?

- duplication
- copy and paste somewhere else (other blocks, entry types and even sections)
- drag and drop from one entry type to another, if the same elements are allowed

**Reply by brandonkelly** [reply]
Each of those are things we want to get to, yeah. Can’t give you an ETA though.

**Reply by vandres** [reply]
Thanks for the reply! I am almost done with the first two features. Could you point me in the right direction for the drag and drop?
